src/coreclr/gc/gc.cpp

```diff
@@ -7699,7 +7699,7 @@ BOOL grow_mark_stack (mark*& m, size_t& len, size_t init_len)
     if (tmp)
     {
         memcpy (tmp, m, len * sizeof (mark));
-        delete m;
+        delete[] m;
```

Comment:
There is another instance that can be added to this PR. Look for `delete mark_stack_array;`. The allocation of that is passed into `make_mark_stack()` and done with a `new[]`.